extractor/src/main/java/org/schabi/newpipe/extractor/search/filter/Filter.java

```diff
@@ -0,0 +1,33 @@
+package org.schabi.newpipe.extractor.search.filter;
+
+public final class Filter {
+
+    public static final int ITEM_IDENTIFIER_UNKNOWN = -1;
+    private final FilterGroup[] sortGroups;
+    private int size = 0;
+
+    private Filter(final FilterGroup[] sortGroups) {
+        this.sortGroups = sortGroups;
+    }
+
+    public FilterGroup[] getFilterGroups() {
+        return sortGroups;
+    }
+
+    public static class Builder {
+        final Filter filter;
+
+        public Builder(final FilterGroup[] sortGroups) {
+            filter = new Filter(sortGroups);
+        }
+
+        public Builder setNoOfFilters(final int noOfFilters) {
+            filter.size = noOfFilters;
+            return this;
+        }
+
+        public Filter build() {
+            return filter;
+        }
+    }
+}
```

Comment:
You're right this thing needs some cleanup. Thank you. The idea of the Builder was. That only the builder can set the size and nobody else, but this is a leftover. I have this pattern with the FilterItem also. Because ther I do not want that the client is able to change - in that case - the identifiers.

Reply:
cleaned